homeassistant/components/netatmo/webhook.py

```diff
@@ -62,21 +65,32 @@ def async_evaluate_event(hass, event_data):
             person_event_data[ATTR_IS_KNOWN] = person.get(ATTR_IS_KNOWN)
             person_event_data[ATTR_FACE_URL] = person.get(ATTR_FACE_URL)
 
-            async_send_event(hass, event_type, person_event_data)
+            await async_send_event(hass, event_type, person_event_data)
 
     else:
-        _LOGGER.debug("%s: %s", event_type, event_data)
-        async_send_event(hass, event_type, event_data)
+        await async_send_event(hass, event_type, event_data)
 
 
 @callback
-def async_send_event(hass, event_type, data):
+async def async_send_event(hass, event_type, data):
     """Send events."""
-    hass.bus.async_fire(
-        event_type=NETATMO_EVENT, event_data={"type": event_type, "data": data}
-    )
+    _LOGGER.debug("%s: %s", event_type, data)
     async_dispatcher_send(
         hass,
         f"signal-{DOMAIN}-webhook-{event_type}",
         {"type": event_type, "data": data},
     )
+
+    if event_type not in EVENT_ID_MAP:
+        return
+
+    hass.bus.async_fire(
+        event_type=NETATMO_EVENT,
+        event_data={
+            "type": event_type,
+            "data": data,
+            ATTR_DEVICE_ID: hass.data[DOMAIN][DATA_DEVICE_IDS].get(
+                data.get(EVENT_ID_MAP.get(event_type))
```

Comment:
This would be easier to follow if `data.get(EVENT_ID_MAP.get(event_type))` was assigned to a variable. It took me a while to track this back to understand what is going on

Reply:
Also so you have a guard above I think you can use `device_id = EVENT_ID_MAP[event_type]`